internal/osbuildexecutor/runner-impl-aws-ec2.go

```diff
@@ -53,52 +95,133 @@ func (ec2e *awsEC2Executor) RunOSBuild(manifest []byte, store, outputDirectory s
+		"-rf",
+		filepath.Join("/var/tmp/osbuild-composer", "input.tar"),
+		store,
 	)
+	if err = cmd.Run(); err != nil {
+		logrus.Errorf("Unable to create input tar: %v", err)
+		return nil, err
+	}
 
-	stdout := &bytes.Buffer{}
-	stderr := &bytes.Buffer{}
-	cmd.Stdout = stdout
-	cmd.Stderr = stderr
-	cmd.Stdin = bytes.NewReader(manifest)
-
-	err = cmd.Start()
+	client := http.Client{
+		Timeout: time.Minute * 60,
+	}
+	inputFile, err := os.Open(filepath.Join("/var/tmp/osbuild-composer", "input.tar"))
 	if err != nil {
-		logrus.Errorf("Starting osbuild-jobsite-manager failed: %v", err)
 		return nil, err
 	}
-	err = cmd.Wait()
+	defer os.Remove(inputFile.Name())
+
+	resp, err := client.Post(fmt.Sprintf("%s:8001/api/v1/build", *si.Instance.PrivateIpAddress), "application/x-tar", inputFile)
 	if err != nil {
-		logrus.Errorf("Waiting for osbuild-jobsite-manager failed: %v", err)
-		if e, ok := err.(*exec.ExitError); ok {
-			logrus.Errorf("Exit code: %d", e.ExitCode())
+		logrus.Errorf("Unable to request build from executor instance: %v", err)
+		return nil, err
+	}
+	if resp.StatusCode != 201 {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logrus.Warningf("Unable to read body waiting for build to run: %v", err)
+			return nil, err
 		}
-		logrus.Errorf("StdErr :%s", stderr.String())
+		logrus.Errorf("Something went wrong during executor build: %s", body)
 		return nil, err
 	}
 
 	var osbuildResult osbuild.Result
-	err = json.Unmarshal(stdout.Bytes(), &osbuildResult)
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(body, &osbuildResult)
+	if err != nil {
+		logrus.Errorf("Unable to unmarshal stdout into osbuild result: %v", err)
+		return nil, err
+	}
+	if !osbuildResult.Success {
+		return &osbuildResult, nil
+	}
+
+	// fetch disk
+	resp, err = client.Get(fmt.Sprintf("%s:8001/api/v1/result/output.tar", *si.Instance.PrivateIpAddress))
+	if err != nil {
+		logrus.Errorf("Unable to get result tarball: %s", err)
+		return &osbuildResult, err
+	}
+	if resp.StatusCode != 200 {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logrus.Warningf("Unable to read body while fetching executor result: %v", err)
+			return nil, err
+		}
+		logrus.Errorf("Something went wrong during executor build: %s", body)
+		return nil, err
+	}
+	file, err := os.Open(filepath.Join(outputDirectory, "output.tar"))
+	if err != nil {
+		logrus.Errorf("Unable to write executor result tarball :%v", err)
+		return nil, err
+	}
+	defer file.Close()
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		logrus.Errorf("Unable to write executor result tarball :%v", err)
+		return nil, err
+	}
+	err = file.Close()
+	if err != nil {
+		logrus.Errorf("Unable to write executor result tarball :%v", err)
+		return nil, err
+	}
+	cmd = exec.Command("tar",
+		"-Sxf",
+		"output.tar",
+	)
+	cmd.Dir = outputDirectory
+	err = cmd.Start()
```

Comment:
Could this be slightly shorter by just using `cmd.Run()` (or `cmd.CombinedOutput()`) ? Something like:
```go
	cmd.Dir = outputDirectory
	if output, err := cmd.CombinedOutput(); err != nil {
		logrus.Errorf("Untarring image failed: %v output:\n%s", err, output)
		return nil, err
	}
```